demosplan/DemosPlanCoreBundle/Logic/Permission/AccessControlService.php

```diff
@@ -209,15 +210,42 @@ public function enablePermissionCustomerOrgaRole(string $permissionToEnable, Cus
                 continue;
             }
 
+            $updatedOrga = $this->addPermissionBasedOnOrgaType($permissionToEnable, $role, $orgaInCustomer, $customer, $dryRun);
+
+            if (null !== $updatedOrga) {
+                $updatedOrgas[] = $updatedOrga;
+            }
+        }
+
+        return $updatedOrgas;
+    }
+
+    /**
+     * $orgatype = getorgatype.
+     *
+     * $orgatypeForRole = getOrgaTypebasedonRole
+     * if $orgatype === $orgatypeForRole --> add permission
+     */
+    private function addPermissionBasedOnOrgaType(string $permissionToEnable, RoleInterface $role, OrgaInterface $orgaInCustomer, CustomerInterface $customer, bool $dryRun): ?OrgaInterface
+    {
+        $orgaTypesInCustomer = $orgaInCustomer->getTypes($customer->getSubdomain(), true);
+        foreach ($orgaTypesInCustomer as $orgaTypeInCustomer) {
+            if (self::CREATE_PROCEDURES_PERMISSION === $permissionToEnable
```

Comment:
This is ok for now to get this done,  but it does not scale very well. We need to find an approach that this check could be done by configuration

Reply:
@mussbach I am with you. What about disccusing in tehnix